foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/Fabric8LocationMigration.java

```diff
@@ -0,0 +1,148 @@
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.boozallen.aissemble.upgrade.migration.v1_8_0.OrphedomosToFabric8Migration.FABRIC8_ARTIFACT_ID;
+import static org.apache.commons.lang3.StringUtils.repeat;
+import static org.technologybrewery.baton.util.pom.LocationAwareMavenReader.END;
+import static org.technologybrewery.baton.util.pom.LocationAwareMavenReader.START;
+
+/**
+ * In the move from Orphedomos to Fabric8, we can no longer specify the Docker plugin as part of the build for a module
+ * that is not a `docker-build` packaging type.  (Orphedomos allowed this and simply skipped all goals if the packaging
+ * type was not `orphedomos`.)  This migration ensures any `pom` type modules only specify the Docker plugin within a
+ * pluginManagement section.
+ */
+public class Fabric8LocationMigration extends AbstractPomMigration {
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        Model model = PomHelper.getLocationAnnotatedModel(file);
+        String packaging = model.getPackaging();
+        if (POM.equals(packaging)) {
+            boolean profilesMatch = model.getProfiles()
+                    .stream()
+                    .map(Profile::getBuild)
+                    .map(Fabric8LocationMigration::getDockerPluginIfPresent)
+                    .anyMatch(Objects::nonNull);
+            return profilesMatch || getDockerPluginIfPresent(model.getBuild()) != null;
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        detectAndSetIndent(file);
+        Model model = PomHelper.getLocationAnnotatedModel(file);
+        PomModifications modifications;
+        modifications = new PomModifications();
+        for (BuildBase build : getAllBuilds(model)) {
+            Plugin dockerPlugin = getDockerPluginIfPresent(build);
+            if (dockerPlugin != null) {
+                movePlugin(modifications, dockerPlugin, build, file);
+            }
+        }
+        return PomHelper.writeModifications(file, modifications.finalizeMods());
+    }
+
+    private void movePlugin(PomModifications modifications, Plugin dockerPlugin, BuildBase build, File file) {
+        List<String> content = getPluginContent(dockerPlugin, file);
+        boolean removePluginsSection = removePluginFromPlugins(modifications, dockerPlugin, build);
+        addPluginToManagement(modifications, build, removePluginsSection, content);
+    }
+
+    private void addPluginToManagement(PomModifications modifications, BuildBase build, boolean removePluginsSection, List<String> content) {
+        InputLocation insertLocation;
+        PluginManagement management = build.getPluginManagement();
+        if (management == null ) {
+            InputLocation pluginsStart = build.getLocation("plugins" + START);
+            insertLocation = removePluginsSection ? build.getLocation(END) : pluginsStart;
```

Comment:
I: Just allows a tidier insertion by putting the new plugin management block above plugins if that section isn't being completely removed.